gc/gc-base/src/main/java/org/projectnessie/versioned/persist/gc/GCImpl.java

```diff
@@ -0,0 +1,130 @@
+    //      on cutoff time of this reference). Add its content to bloom filter.
+    //    Else if the content is expired, Stop traversing the commits
+    //      as there cannot be any more live commits for this reference.
+    //    Collect bloom filter per content id from each task and merge them.
+    try (NessieApiV1 api = DistributedIdentifyUtil.getApi(configuration)) {
+      List<Reference> references = api.getAllReferences().get().getReferences();
+      DistributedIdentifyUtil distributedIdentifyUtil =
+          new DistributedIdentifyUtil(session, configuration);
+      Map<String, ContentBloomFilter> liveContentsBloomFilterMap =
+          distributedIdentifyUtil.getLiveContentsBloomFilters(
+              references, cutOffTimestampPerRef, defaultCutOffTimeStamp);
+
+      // b) Algorithm for identifying the expired and live contents
+      //      and return the list of expired contents and list of live contents per content id.
+      //    Get the list of dead reference from reflog
+      //      and append it to current list of live references.
+      //    Walk through each reference distributively (one spark task for each reference).
+      //    For each commit in a reference,
+      //      get the contents (use DETACHED reference here as some references are dead).
+      //    Check if the content is expired by checking against the bloom filter.
+      //    Merge the result of each task and return the result.
+      List<Reference> deadReferences = collectDeadReferences(api);
+      List<Reference> allRefs = new ArrayList<>(references);
+      if (deadReferences.size() > 0) {
+        allRefs.addAll(deadReferences);
+      }
+      return distributedIdentifyUtil.getIdentifiedResults(liveContentsBloomFilterMap, allRefs);
+    }
+  }
+
+  private List<Reference> collectDeadReferences(NessieApiV1 api) {
+    List<RefLogResponse.RefLogResponseEntry> reflogResponseEntries;
+    try {
+      reflogResponseEntries =
+          StreamingUtil.getReflogStream(
+                  api,
+                  null,
+                  null,
+                  "reflog.operation == 'DELETE_REFERENCE' || reflog.operation == "
+                      + "'ASSIGN_REFERENCE'",
+                  OptionalInt.empty())
+              .collect(Collectors.toList());
+    } catch (NessieNotFoundException e) {
+      throw new RuntimeException(e);
+    }
+    return reflogResponseEntries.stream()
+        .map(
+            entry -> {
+              String hash;
+              if (entry.getOperation().equals("DELETE_REFERENCE")) {
+                hash = entry.getCommitHash();
+              } else if (entry.getOperation().equals("ASSIGN_REFERENCE")) {
+                hash = entry.getSourceHashes().get(0);
+              } else {
+                throw new RuntimeException(
+                    entry.getOperation() + " operation found in dead reflog query");
+              }
+              if (entry.getRefType().equals("Branch")) {
```

Comment:
Actually, this code construct should reuse functionality in `RefUtil`

Reply:
for that GC need to depend on one more module, so added constant in `RefLogResponseEntry`